### PR TITLE
Upgrade codeflare-sdk 0.18.0 for py3.9 content

### PR DESCRIPTION
**Pull request by harshad16**
## Description

Upgrade codeflare-sdk 0.18.0 for py3.9 content
Fixes: https://github.com/project-codeflare/codeflare-sdk/actions/runs/10114395555/job/27972928826

Neglected:
- Python 3.8-based content as codeflare-sdk 0.16.3 was the last supported version.
- Intel content as it is currently unmaintained.

## How Has This Been Tested?

CI tests

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by jstourac** [comment]
Python package diff for the `rocm-jupyter-pytorch-ubi9-python-3.9` image build from main vs from this PR:

```
$ diff -u0 pip_pr pip
--- pip_pr	2024-07-31 12:52:41.000000000 +0200
+++ pip	2024-07-31 12:52:57.000000000 +0200
@@ -15 +15 @@
-astroid                   3.2.4
+astroid                   3.2.2
@@ -21 +21 @@
-bcrypt                    4.2.0
+bcrypt                    4.1.3
@@ -25,5 +25,5 @@
-bokeh                     3.4.3
-boto3                     1.34.150
-botocore                  1.34.150
-cachetools                5.4.0
-certifi                   2024.7.4
+bokeh                     3.4.2
+boto3                     1.34.135
+botocore                  1.34.135
+cachetools                5.3.3
+certifi                   2024.6.2
@@ -33 +33 @@
-codeflare-sdk             0.18.0
+codeflare-sdk             0.16.4
@@ -52 +52 @@
-exceptiongroup            1.2.2
+exceptiongroup            1.2.1
@@ -57 +57 @@
-fonttools                 4.53.1
+fonttools                 4.53.0
@@ -65 +65 @@
-google-auth               2.32.0
+google-auth               2.30.0
@@ -67 +67 @@
-google-cloud-storage      2.18.0
+google-cloud-storage      2.17.0
@@ -71 +71 @@
-grpcio                    1.65.1
+grpcio                    1.64.1
@@ -73 +73 @@
-importlib_metadata        8.2.0
+importlib_metadata        8.0.0
@@ -75 +75 @@
-ipykernel                 6.29.5
+ipykernel                 6.29.4
@@ -87 +87 @@
-jsonschema                4.23.0
+jsonschema                4.22.0
@@ -107 +107 @@
-jupyterlab_server         2.27.3
+jupyterlab_server         2.27.2
@@ -158 +158 @@
-pillow                    10.4.0
+pillow                    10.3.0
@@ -166 +166 @@
-protobuf                  4.25.4
+protobuf                  4.25.3
@@ -168 +168 @@
-psycopg                   3.1.20
+psycopg                   3.1.19
@@ -170 +170 @@
-pure_eval                 0.2.3
+pure-eval                 0.2.2
@@ -172 +172 @@
-pyarrow                   17.0.0
+pyarrow                   16.1.0
@@ -184 +184 @@
-pylint                    3.2.6
+pylint                    3.2.4
@@ -198 +198 @@
-ray                       2.23.0
+ray                       2.20.0
@@ -207 +207 @@
-rpds-py                   0.19.1
+rpds-py                   0.18.1
@@ -225 +225 @@
-sympy                     1.13.1
+sympy                     1.12.1
@@ -227 +227 @@
-tenacity                  9.0.0
+tenacity                  8.4.2
@@ -236 +236 @@
-tomlkit                   0.13.0
+tomlkit                   0.12.5
@@ -255 +255 @@
-whatthepatch              1.0.6
+whatthepatch              1.0.5
```

The diff seems okay to me.

Also, all the relevant files are updated in this PR, so also good.

I would consider to have the dependency fix in a separate commit than the actual codeflare-sdk upgrade. But I can live with that.

/lgtm

**Comment by harshad16** [comment]
>I would consider to have the dependency fix in a separate commit than the actual codeflare-sdk upgrade. But I can live with that.

As we would have to retrigger a workflow , which isn't on our end, this is more of convenience way.

**Comment by harshad16** [comment]
Thanks for the review folks.

/override ci/prow/rocm-notebooks-e2e-tests
/override ci/prow/images
/override ci/prow/notebooks-ubi9-e2e-tests
/override ci/prow/runtimes-ubi9-e2e-tests

File: jupyter/rocm/pytorch/ubi9-python-3.9/Pipfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ name = "pytorch"
 tensorboard = "~=2.16.2"
 torch = {version = "~=2.3.1", index = "pytorch"}
 torchvision = {version = "~=0.18.1", index = "pytorch"} # https://pytorch.org/get-started/previous-versions/#linux-and-windows-1
+pytorch-triton-rocm = {version = "~=2.3.1", index = "pytorch"}
```
Comment (jstourac) [inline jupyter/rocm/pytorch/ubi9-python-3.9/Pipfile:18]: I just realized that this added line should be added also into the `runtimes/rocm-pytorch/ubi9-python-3.9/Pipfile`. I addressed this during some other work in #656. Shall we amend this PR with this addition before this is merged or are we okay with the fix in the followup PR? :thinking:

**Pull request merged**